datahub/opal.py

```diff
@@ -61,9 +109,17 @@ def create_opal_frame() -> pd.DataFrame:
     return df
 
 
-def append_opal_frame(data: list[float]) -> pd.DataFrame:
+def append_opal_frame(data: dict[str, float]) -> pd.DataFrame:
     """Function that creates pandas data frame for Opal data to be appended."""
-    df = pd.DataFrame([data], columns=opal_header)
+    data_array = []
+
+    for item in key_headers:
+        if item in data:
+            data_array.append(data[item])
+        elif item == "N/A":
+            data_array.append(0)
+
+    df = pd.DataFrame([data_array], columns=opal_header)
```

Comment:
We want to return a series now

```suggestion
    row = pd.Series(data_array, name=data["frame"], index=opal_header)
```

The lines below this use the `df` variable name - this should change to `row` now